shared/question/question.ts

```diff
@@ -181,6 +194,99 @@ class DateQuestion extends Question {
     }
 }
 
+class MultipleChoiceTableQuestion extends Question {
+    /**
+     * table is from one of the following types: MULTIPLE_CHOICE_TABLE, CHECKBOX_TABLE, TEXT_TABLE
+     * MULTIPLE_CHOICE_TABLE: each row represents a multipleChoice question, where the options are the columns.
+     * answer is array of undefined\picked column index, size: (rows x 1)
+     * @param columns: array of columns titles
+     * @param rows: array of rows titles
+     */
+    columns: string[];
+
+    rows: RowTable[];
+
+    constructor(questionObject: IQuestion) {
+        super(questionObject);
+        this.columns = questionObject.columns;
+        this.rows = questionObject.rows;
+    }
+
+    isAnswerValid() {
+        if (this.answer.length !== this.rows.length) return false;
+        const answerValidation = this.rows.map((row, idx) => {
+            if (this.answer[idx] < 0 || this.answer[idx] > this.columns.length - 1 || (row.required && this.answer[idx] === undefined)) {
+                return false;
+            }
+            return true;
+        });
+        return !answerValidation.includes(false) && super.isAnswerValid();
+    }
```

Comment:
Use `Array.prototype.every` instead. It's a significantly faster method.

```suggestion
        return this.rows.every((row, idx) => this.answer[idx] > 0 && this.answer[idx] < this.columns.length  && (!row.required || this.answer[idx])) && super.isAnswerValid();
    }
```